tests/ga/test_policy_engine.py

```diff
@@ -15,20 +15,376 @@
 # Requires Python 2.4+ and Openssl 1.0+
 #
 
+import os
+import json
+
 from tests.lib.tools import AgentTestCase
-from azurelinuxagent.ga.policy.policy_engine import PolicyEngine
-from tests.lib.tools import patch
+from azurelinuxagent.ga.policy.policy_engine import ExtensionPolicyEngine
+from tests.lib.tools import patch, data_dir
+from azurelinuxagent.common.protocol.restapi import Extension
+
+TEST_EXTENSION_NAME = "Microsoft.Azure.ActiveDirectory.AADSSHLoginForLinux"
+
+
+class TestExtensionPolicyEngine(AgentTestCase):
+    def setUp(self):
+        self.custom_policy_path = os.path.join(data_dir, 'ga', "waagent_policy.json")
 
+        # Patch attributes to enable policy feature
+        self.patch_custom_policy_path = patch('azurelinuxagent.ga.policy.policy_engine.CUSTOM_POLICY_PATH',
+                                              new=self.custom_policy_path)
+        self.patch_custom_policy_path.start()
+        self.patch_conf_flag = patch('azurelinuxagent.ga.policy.policy_engine.conf.get_extension_policy_enabled',
+                                     return_value=True)
+        self.patch_conf_flag.start()
+
+        AgentTestCase.setUp(self)
+
+    def tearDown(self):
+        # Clean up any custom policy file we created
+        if os.path.exists(self.custom_policy_path):
```

Comment:
isn't this deleting the data file from the source code?

Reply:
or maybe you meant to use self.tmp_dir

Reply:
We don't actually store a test policy file in the source code, each test creates its own file and cleans it up. But I was looking for a better place to create this file, so I've changed it to use self.tmp_dir.

Reply:
tearDown cleans up self.tmp_dir, no need for explicit deletion

Reply:
Removed